addon/v1alpha1/types_managedclusteraddon.go

```diff
@@ -280,7 +268,134 @@ const (
 	// HostedManifestLocationHostingValue is a value of the annotation HostedManifestLocationAnnotationKey,
 	// indicates the manifest will be deployed on the hosting cluster in Hosted mode.
 	HostedManifestLocationHostingValue = "hosting"
-	// HostedManifestLocationNoneValue is a value of the annotation HostedManifestLocationAnnotationKey,,
+	// HostedManifestLocationNoneValue is a value of the annotation HostedManifestLocationAnnotationKey,
 	// indicates the manifest will not be deployed in Hosted mode.
 	HostedManifestLocationNoneValue = "none"
+
+	// Finalizers on the managedClusterAddon.
+
+	// AddonDeprecatedPreDeleteHookFinalizer is the finalizer for an addon which has deployed hook objects.
+	// Deprecated: please use the AddonPreDeleteHookFinalizer instead.
+	AddonDeprecatedPreDeleteHookFinalizer = "cluster.open-cluster-management.io/addon-pre-delete"
```

Comment:
why we add deprecated?

Reply:
I want to use `addon.open-cluster-management.io/addon-pre-delete` instead of `cluster.open-cluster-management.io/addon-pre-delete`

Reply:
do we need to put deprecated in the api repo given they are deprecated?